lib/monitor/run.js

```diff
@@ -16,6 +16,27 @@ var psTree = require('pstree.remy');
 var path = require('path');
 var signals = require('./signals');
 
+function waitForOldKids(oldKids, callback) {
+  if (oldKids) {
+    // check if all previous kids have been terminated
+    exec('kill -0  ' + oldKids.join(' '), (error) => {
+      const returnCode = error ? error.code : 0;
+      if (returnCode < 126) { // ignore command not found error
+        const stillRunningKids = oldKids.length - returnCode;
```

Comment:
When I run `kill` it always returns `0` or `1` so I'm not sure if this math is reliable.  You might have to run psTree again for each child to see whether it is still running and whether it has added any new child processes.

Reply:
Maybe what you could do is just poll `kids[0]`.  If it isn't running any more, call `waitForOldKids(oldKids.slice(1), callback)`.  Then when `oldKids.length === 0` you are done.

Reply:
Also, using `psTree` to wait for the child process might work in windows as well, where `kill` is not available.

Reply:
Actually since `psTree` doesn't return the pid given as part of the list, you can't use it to tell whether a process is still running.

Reply:
In this part of code I only want to wait for the subprocesses to terminate. I think using pstree here instead of `kill -0` is a good idea.

Reply:
Oh yeah, if you pass in the root pid you can just us pstree to check if there are any children or not instead of trying to kill each child.